www/js/controllers.js

```diff
@@ -1,44 +1,41 @@
 'use strict';
 
 import angular from 'angular';
+import { addStatError, addStatReading, statKeys } from './plugin/clientStats';
 
 angular.module('emission.controllers', ['emission.splash.startprefs',
                                         'emission.splash.pushnotify',
                                         'emission.splash.storedevicesettings',
                                         'emission.splash.localnotify',
-                                        'emission.splash.remotenotify',
-                                        'emission.stats.clientstats'])
+                                        'emission.splash.remotenotify'])
 
 .controller('RootCtrl', function($scope) {})
 
 .controller('DashCtrl', function($scope) {})
 
 .controller('SplashCtrl', function($scope, $state, $interval, $rootScope, 
     StartPrefs, PushNotify, StoreDeviceSettings,
-    LocalNotify, RemoteNotify, ClientStats)  {
+    LocalNotify, RemoteNotify)  {
   console.log('SplashCtrl invoked');
   // alert("attach debugger!");
   // PushNotify.startupInit();
 
   $rootScope.$on('$stateChangeSuccess', function(event, toState, toParams, fromState, fromParams){
     console.log("Finished changing state from "+JSON.stringify(fromState)
         + " to "+JSON.stringify(toState));
-    ClientStats.addReading(ClientStats.getStatKeys().STATE_CHANGED,
-      fromState.name + '-2-' + toState.name).then(function() {}, function() {});
+    addStatReading(statKeys.STATE_CHANGED, fromState.name + '-2-' + toState.name);
```

Comment:
I think we should be able to remove `controllers.js` now, along with `main.js`.

> shouldn't we implement the equivalent in the react routing framework?

Yes, we most definitely should. But I think this would be easiest to re-implement client stats once onboarding is stable and finalized, and those changes are pulled into this branch `service_rewrite_2023`.

Reply:
Retaining this as a **future fix**